skyd/cache.go

```diff
@@ -0,0 +1,175 @@
+package skyd
+
+import (
+	"sync"
+
+	"gitlab.com/NebulousLabs/errors"
+	"gitlab.com/SkynetLabs/skyd/build"
+	"gitlab.com/SkynetLabs/skyd/node/api"
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+)
+
+type (
+	// PinnedSkylinksCache is a simple cache of the renter's directory
+	// information, so we don't need to fetch that for each skylink we
+	// potentially want to pin/unpin.
+	PinnedSkylinksCache struct {
+		result   *RebuildCacheResult
+		skylinks map[string]struct{}
+		mu       sync.Mutex
+	}
+	// RebuildCacheResult informs the caller on the status of a cache rebuild.
+	// The error should not be read before the channel is closed.
+	RebuildCacheResult struct {
+		// Ch indicates the status of the cache rebuild progress - if it's not
+		// closed then the rebuild is still in progress.
+		Ch chan struct{}
```

Comment:
(lord of the nits) f/u You could make this a `<-chan struct{}` to prevent the receiver of the result from closing the channel.
`ErrAvail` would be a nice name I think. I thought we usually call them `c` so I had to double check but apparently we commonly use nice names.